backends-common/rabbitmq/src/test/java/org/apache/james/backends/rabbitmq/RabbitMQTest.java

```diff
@@ -327,6 +330,137 @@ void rabbitMQShouldSupportTheExclusiveWorkQueueCase() throws Exception {
+                    (consumerTag, message) -> {
+                        try {
+                            firstRegisteredConsumerMessageCount.incrementAndGet();
+                            TimeUnit.SECONDS.sleep(1);
+                            channel2.basicAck(message.getEnvelope().getDeliveryTag(), false);
+                        } catch (InterruptedException e) {
+                            //do nothing
+                        }
+                    },
+                    (consumerTag -> { }));
+                channel3.basicConsume(WORK_QUEUE, !AUTO_ACK, "starvingConsumer", !NO_LOCAL, !EXCLUSIVE, arguments,
+                    (consumerTag, message) -> {
+                        try {
+                            secondRegisteredConsumerMessageCount.incrementAndGet();
+                            TimeUnit.SECONDS.sleep(1);
+                            channel3.basicAck(message.getEnvelope().getDeliveryTag(), false);
+                        } catch (InterruptedException e) {
+                            //do nothing
+                        }
+                    },
+                    consumerTag -> { });
+
+                awaitAtMostOneMinute.until(() -> (firstRegisteredConsumerMessageCount.get() + secondRegisteredConsumerMessageCount.get()) == 10);
+
+                assertThat(firstRegisteredConsumerMessageCount.get()).isEqualTo(10);
+                assertThat(secondRegisteredConsumerMessageCount.get()).isEqualTo(0);
+            }
+
+            @Test
+            void rabbitMQShouldProvideSingleActiveConsumerName() throws Exception {
+                channel1.exchangeDeclare(EXCHANGE_NAME, "direct", DURABLE);
+                channel1.queueDeclare(WORK_QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, Constants.WITH_SINGLE_ACTIVE_CONSUMER);
+                channel1.queueBind(WORK_QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+                channel1.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, "foo".getBytes(StandardCharsets.UTF_8));
+
+                AtomicInteger deliveredMessagesCount = new AtomicInteger(0);
+
+                String firstRegisteredConsumer = "firstRegisteredConsumer";
+                ImmutableMap<String, Object> arguments = ImmutableMap.of();
+                channel2.basicConsume(WORK_QUEUE, AUTO_ACK, firstRegisteredConsumer, !NO_LOCAL, !EXCLUSIVE, arguments,
+                    (consumerTag, message) -> deliveredMessagesCount.incrementAndGet(),
+                    (consumerTag -> { }));
+                channel3.basicConsume(WORK_QUEUE, AUTO_ACK, "starvingConsumer", !NO_LOCAL, !EXCLUSIVE, arguments,
+                    (consumerTag, message) -> deliveredMessagesCount.incrementAndGet(),
+                    consumerTag -> { });
+
+                awaitAtMostOneMinute.until(() -> deliveredMessagesCount.get() > 0);
+                awaitAtMostOneMinute.until(() -> rabbitMQExtension.managementAPI()
```

Comment:
So i really try to reproduce my initial problems with no avail.
I think we could consider it stable. And react if it starts failing on the CI.

For the other test ( in  the distributed task manager), i still don't want to use the management API because as i describe in the corresponding thread, we want to retrieve the node to get the taskmanager which became the single active consumer. And the rabbitmq management api give us the consumer tag which we do not expose from the workqueue nor taskmanager so it can't help us to retrieve the active taskmanager in our test.